internal/mode/static/telemetry/collector.go

```diff
@@ -471,37 +473,41 @@ func parseSnippetValueIntoDirectives(snippetValue string) []string {
 	return directives
 }
 
-// parseContextDirectiveMapIntoLists returns two same-length lists where the elements at each corresponding index
+// parseDirectiveContextMapIntoLists returns two same-length lists where the elements at each corresponding index
 // are paired together.
-// The first list contains strings which are the NGINX context and directive of a Snippet joined with a hyphen.
-// The second list contains ints which are the count of total same context-directive values of the first list.
-// Both lists are ordered based off of count first, then lexicographically on the context-directive string.
-func parseContextDirectiveMapIntoLists(contextDirectiveMap map[sfContextDirective]int) ([]string, []int64) {
-	type sfContextDirectiveCount struct {
-		contextDirective string
-		count            int64
+// The first list contains strings which are the NGINX directive and context of a Snippet joined with a hyphen.
+// The second list contains ints which are the count of total same directive-context values of the first list.
+// Both lists are ordered first by count, then by lexicographical order of the context string,
+// then lastly by directive string.
+func parseDirectiveContextMapIntoLists(directiveContextMap map[sfDirectiveContext]int) ([]string, []int64) {
+	type sfDirectiveContextCount struct {
+		directive, context string
+		count              int64
 	}
 
-	kvPairs := make([]sfContextDirectiveCount, 0, len(contextDirectiveMap))
+	kvPairs := make([]sfDirectiveContextCount, 0, len(directiveContextMap))
 
-	for k, v := range contextDirectiveMap {
-		kvPairs = append(kvPairs, sfContextDirectiveCount{k.context + "-" + k.directive, int64(v)})
+	for k, v := range directiveContextMap {
+		kvPairs = append(kvPairs, sfDirectiveContextCount{k.directive, k.context, int64(v)})
 	}
 
 	sort.Slice(kvPairs, func(i, j int) bool {
 		if kvPairs[i].count == kvPairs[j].count {
-			return kvPairs[i].contextDirective < kvPairs[j].contextDirective
+			if kvPairs[i].context == kvPairs[j].context {
+				return kvPairs[i].directive < kvPairs[j].directive
+			}
+			return kvPairs[i].context < kvPairs[j].context
```

Comment:
should we be sorting by directive first, then context?

Reply:
I contemplated this, I think it would be beneficial to sort by context, then by directive. So for example if everything were at the same count you'd see all the contexts grouped together:

```
directive1-main
directive2-main
directive3-main
directive4-http
directive5-http
directive6-http
.
.
.
```

Reply:
That's fair, it's not a huge deal either way